### DISCUSSION TITLE
Is there a Description of an Interface of SAF to Enterprise Frameworks?

### DISCUSSION THREAD

**Discussion by haarer** [GfSE/SAF-Specification#31] [Q&A] [answered]
From a mail to safsupport
"
Sehr geehrte SAF Arbeitsgruppe,

leider war mir aus Ihren Beschreibungen zum System Architecture Framework in github, die Schnittstelle zu denn Enterprise Frameworks wie NAF oder ADMBw nicht klar ersichtlich.

Ist hier im SAF ein separater Übergabepunkt von Informationen aus der Enterprise Architecture definiert oder ist diese über die Überschneidung durch die „Operational Domain SOV“ Ebene im Viewpoint Grid realisiert?
"

**Comment by haarer** [comment, marked as answer]
Die Anbindung von SAF an Enterprise Frameworks ist so gedacht, daß man die SAF Operational Domain  wegläßt und statt der Relationen zur OD dann Relationen zu den analogen Elementen in UAF / NAF / ADMBw herstellt.
Da in Modellen der o.g. Enterprise Frameworks i.d.R. schon eine "Idee" der zu beschaffenden Systeme vorliegt, sollten diese auch über Relationen mit den entsprechenden konzeptionellen Systeme aus der FD in SAF verbunden werden.
Die Vorgehensweise und die Relationen wären noch im Detail zu beschreiben.

**Reply by parkaneric** [reply]
Aus SAF Sicht könnten folgende SAF Stereotypen zu operationellen Stereotypen in Enterprise Frameworks verknüpft werden: SAF_SystemUseCase (SFV01a), 2. SAF_SystemCapability (SFV02b), SAF_SystemRequirement (SFV06a).

Das würde bspw. für ADMBw@NAF folgende Traceability bedeuten:
1. SAF_SystemUseCase enables OperationalArchitecture (L2)
2. SAF_SystemCapability mapping to Capability (C1)
3. SAF_SystemRequirement derived from Requirement (R2)

Würde man in den Werkzeugen dann einem Element zwei Stereotypen (SAF + NAF) vergeben, um z.B. Verlinkungsregeln zu genügen?

**Reply by parkaneric** [reply]
So in etwa ... falls die Metatypen nicht passen (Use Case vs Class), muss wohl ein Platzhalter auf SAF Basis erstellt werden?

![NAF C1](https://github.com/GfSE/SAF-Specification/assets/105344662/c571410d-8f6b-41de-9720-210015c1defc)
![NAF L2](https://github.com/GfSE/SAF-Specification/assets/105344662/fb1b4c9e-4de8-4c3c-a3ea-0261493da86f)
![ADMBw R2](https://github.com/GfSE/SAF-Specification/assets/105344662/fcaccc11-b79d-438a-b22e-92fe80fe94a6)

**Reply by haarer** [reply]
Ich würde ungern das Enterprise Architektur (EA)Modell verändern um darauf tracen zu können.
Auch wenn es eine einfache Möglichkeit ist.
Lieber würde ich den SAF Tracing Relationen erlauben, auch auf Elemente des EA Modells zu verweisen. 
Also statt in einem UAF Modell den UAF::Capability elementen auch den SAF_OperationalCapability stereotypen zu verpassen, lieber mit SAF_SystemCapabilityMapping direkt auf UAF:: Capability Elemente verweisen.

Ich würde bei einem EA Modell allerdings eher erwarten, dass es bis auf die resource Ebene runter geht, und man aus der SAF FD auf die Resource Ebene tracen sollte.

Es wird da verschiedene Anwendungsfälle geben.

**Reply by odungern** [reply]
Aus meiner Perspektive: Bitte *niemals* einem Element zwei Stereotypen geben, weil automatische Modelltransformation in vielen Fällen unmöglich wird. Oskar

**Comment by odungern** [comment]
hallo, allerseits,
... lese die Diskussionen nur am Rande mit, habe leider keine Kraft übrig, um mich auch hier fundiert einzumischen. Doch erscheint es mir, als wenn zukünftige Anwender sehr viel Ausbildung genießen und konzeptionelle Trennschärfe entwickeln müssen, um SAF (und ggf. die anderen Frameworks) einigermaßen richtig anwenden zu können. Bestimmte Projekte erfordern es sicher, doch zu viele System Architects scheitern bereits an einfacheren Themen und erzeugen Modelle, die eingehend erläutert werden müssen. In der Software und der System Architecture sind wir noch weit entfernt von der Eindeutigkeit eines elektrischen Schaltplans. Habe hierfür auch keine Lösung (bitte nicht falsch verstehen). Vielleicht gelingt es eher mit vereinfachten und vor allem konsistenten Konzepten - das leistet SysML nicht, wie wir alle wissen, und leider auch nicht SysML v2, wie wir gestern im PLM4MBSE diskutiert haben.
Viele Grüße und wirklich von Herzen viel Erfolg - das Thema können wir gern einmal gemeinsam beleuchten, wenn Ihr mögt.
Oskar

**Comment by parkaneric** [comment]
Danke für die Antworten.
Die Verlinkung funktioniert dann bspw. wie unten angezeigt. Im Enterprise Architect (ich nehme mal stark an der Anfragensteller hat den Sparx EA im Einsatz?) darf man dann nicht die SAF Views nutzen, damit man die SAF Elemente mit ADMBw Elementen verknüpfen kann, da in den Views die Verlinkungsregeln hinterlegt sind. 

Generell gesprochen: brauchen wir irgendwo eine Art How-To?

![ADMBw-SAF](https://github.com/GfSE/SAF-Specification/assets/105344662/023ae8a9-ea2a-400e-8fc0-fa97922415d1)

**Comment by haarer** [comment]
Am 25. Oktober 2023 08:18:23 MESZ schrieb Eric Parkan ***@***.***>:
>Danke für die Antworten.
>Die Verlinkung funktioniert dann bspw. wie unten angezeigt. Im Enterprise Architect (ich nehme mal stark an der Anfragensteller hat den Sparx EA im Einsatz?) darf man dann nicht die SAF Views nutzen, damit man die SAF Elemente mit ADMBw Elementen verknüpfen kann, da in den Views die Verlinkungsregeln hinterlegt sind. 
>
>Generell gesprochen: brauchen wir irgendwo eine Art How-To?
>
>![ADMBw-SAF](https://github.com/GfSE/SAF-Specification/assets/105344662/023ae8a9-ea2a-400e-8fc0-fa97922415d1)
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/GfSE/SAF-Specification/discussions/31#discussioncomment-7377429
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
SAF Viewpoints spezifizieren was in Views drin sein soll, damit die concerns erfüllt sind.

Das heißt nicht, das man da nicht auch weitere Elemente rein tun darf.

Ich könnte mir vorstellen, das man in eine n SAF View auch admbw Elemente aufnimmt und die concerns so erfüllt.

Das (Luxus) Problem hierbei könnte sein, das 
1. eine Metrik/Validierung die man für SAF gemacht hat, nicht erkennt das der VP OK ist, weil die Regeln die SAF fremden Elemente nicht kennen.

2. Viewpoints die von solchen hybrid befüllten VP abhängen, möglicherweise nicht funktionieren, weil dort nicht echte SAF Elemente drin sind.

Beides sind Fragen die in erster Linie das Profildesign betreffen.

Allerdings könnte man überlegen, ob man die Anbindung an EA Fw.  auf eine formal sichere Basis stellt, indem man z.B. spezifiziert, welche stereotypen äquivalent sind.

Btw. Admb:: OperationalArchitekture ist nicht äquivalent zum Usecase. Das ist eher wie der operational context.


-- 
Gruß,
Alexander

**Comment by hugoormo** [comment]
Je nach dem was man als Enterprise Framework versteht. Z.B. im Fall TOGAF geht es um zwei unterschiedliche Anwendungsperspektive: bei SAF geht es um alle Arten von Systemen und bei TOGAF speziell um IT-Systeme.

**Reply by haarer** [reply]
**SAF** ist in dem Sinne kein EA Framework, weil es nicht primär für die Beschreibung von Enterprise Architekturen gedacht ist.
Es ist für die Systementwicklung gedacht. Es hat eine OD, um den Informationsbedarf  der sonst typischerweise von EA Fw befriedigt wird, aus Sicht der Systementwicklung zu befriedigen.

**EA Frameworks wie NAF,ADMBw, UAF, DoDAF, TOGAF**  sind nicht primär für die Systementwickung (IT oder andere) gedacht, sondern für die Beschreibung von Enterprises. Sie haben jeweils eine Ebene die  technische Systeme beschreibt, mit dem Ziel ihre Beschaffung zu planen. (aber nicht mit dem Ziel sie mittels des Frameworks zu entwickeln)

Dass in der Praxis Auftraggeber hier oft über das Ziel hinaus schießen und versuchen "in die Systeme hinein zu designen"- traurig aber wahr.

**Reply by noltem** [reply]
Hier hätte ich mal eine etwas naive, vlt auch akademisch hypothetische Interessensfrage: Könnte es, wenn es sich bei AG und AN um die gleiche Organisation handelt, letzteres nicht ein Grund sein, dass die Operational Domains aus SAF und z.B. UAF doch einen gewissen Überlapp haben? (Organisation möchte durch eigene Systeme z.B. zusätzliche Capabilities fielden, um Unternehmungen "besser", "schneller", "effizienter" durchführen zu können) - Auch ohne dass ein AG über das Ziel hinausschießt?

Könnte evtl. die Notwendigkeit, Systeme für verschiedene Phasen einer Enterprise zu entwickeln dazu führen, dass aus Organisationssicht ein UAF High Level Operational Concept formuliert wird, in dem eine grobe Idee von benötigten Operational Performern skizziert wird, die ggf. für mehrere Enterprise Phases relevant sind. - Als Systementwickler interessiert mich dann eine konkrete Operational Architecture, deren Operational Performer z.B. in einem konkreten SAF Operational Context detailliert werden können. 
Zumal (persönliche Meinung) aus *S*E-Sicht, die Operationale Domäne eine angenehem reduzierte Komplexität besitzt...

Die Relationen zwischen UAF und SAF wären dann natürlich etwas genauer zu diskutieren. Zumindest, wenn man nicht alle Elemente doppelt modellieren möchte: Bei SAF System und Operational Capabilities hätte ich *semantisch* jetzt z.B. keine größeren Bauchschmerzen, sie als Spezialisierungen von UAF Capabilities mit Default Tags (kind: Operational, custom kind: System) zu sehen.
Bei SAF Operational Performer - spezialisiert -> UAF Operational Performer hätte ich etwas größere Schmerzen, dafür sind die Konzepte eigentlich "zu gleich". Das hätte allerdings den "Vorteil", dass man auf Enterprise Ebene in den High Level Operational Concepts schon vorsehen könnte, dass mit SAF "weitermodelliert" wird. - Die Profilmodifikationen, die man in dem Fall vornehmen müsste, wären minimal.

Ich lasse mich aber auch gerne belehren, warum das nicht sinnvoll / gewollt / großer Quatsch / arguing semantics ist.

**Comment by parkaneric** [comment]
Danke für die Rückmeldungen. Die im Paper vorgestellten Konzepte wurden initial untersucht. Folgende Einschätzung:

1. Wenn im Projektkontext zwei organisatorische Entitäten miteinander bspw. im AG/AN Verhältnis stehen, sollten die Enterprise Architektur (EA) und Systemarchitektur (SA) Modelle getrennt bleiben.
2. Eine Verlinkung zwischen EA-SA Elementen ist empfehlenswert. Enge oder lose Kopplung ist abhängig vom Projektkontext.
3. Neben einer Verlinkung von SA-Elementen auf EA-Elemente können auch zusätzliche aus Bottom-Up-Sicht SAF basierte Elemente hinzugefügt werden, welche inhaltlich nicht zum EA-Modell gehören (bspw. zusätzliche Operational Stories als alternative operationelle Anwendung)
4. Solution 1 aus „Linking UAF and SysML Models“ ist Out of Scope bzw. Ist auch nicht empfehlenswert.
5. Solution 2 aus nachtweilhaft, wenn im Projektkontext ein organisatorischer Bruch existiert (AG/AN) – Need-to-know-Prinzip wird verletzt.
6. Solution 3 ist zweckmäßig und sollte weiter untersucht werden.
7. Solution 4 ist häufig beobachtete Praxis in Projekten, folgt aber keinem übergeordneten Architekturziel und etabliert keine zusammenhängende EA-SA.

**Reply by haarer** [reply]
Auf dem TdSE 2025 stellen wir einen Vorschlag vor, wie Enterprise Frameworks wie NAF, ADMBw oder UAF mit dem SAF verbunden werden können. Dazu wird es einen eigenen Viewpoint geben der die Verbindung herstellt.
